File: Simple connection methods for MS-Access using OleDb data provider/[C#][VB]-Simple connection methods for MS-Access using OleDb data provider/C# and VB/Example_vb/DataOperations.vb
Imports System.Data.OleDb
Imports AccessConnections_vb
Public Class DataOperations
    Private fileName As String = ""
    Public Sub New(ByVal databaseFileName As String)
        fileName = databaseFileName
        CustomerTable = New DataTable
    End Sub
    Public Property CustomerTable As DataTable
    Public Sub LoadCustomers()
        Using cn As New OleDbConnection With {.ConnectionString = fileName.BuildConnectionString}
            Using cmd As New OleDbCommand With {.Connection = cn}
                cmd.CommandText = "SELECT CompanyName FROM Customer;"
                Dim dt As New DataTable With {.TableName = "Customer"}
                cn.Open()
                CustomerTable.Load(cmd.ExecuteReader)
            End Using
        End Using
    End Sub
End Class

File: Simple connection methods for MS-Access using OleDb data provider/[C#][VB]-Simple connection methods for MS-Access using OleDb data provider/C# and VB/Example_cs/DataOperations.cs
using AccessConnections_cs;
using System.Data;
using System.Data.OleDb;


namespace Example_cs
{
    public class DataOperations
    {
        private string fileName = "";
        public DataOperations(string databaseFileName)
        {
            fileName = databaseFileName;
            CustomerTable = new DataTable();
        }
        public DataTable CustomerTable { get; set; }
        public void LoadCustomers()
        {
            using (OleDbConnection cn = new OleDbConnection { ConnectionString = fileName.BuildConnectionString() })
            {
                using (OleDbCommand cmd = new OleDbCommand { Connection = cn })
                {
                    cmd.CommandText = "SELECT CompanyName FROM Customer;";
                    DataTable dt = new DataTable { TableName = "Customer" };
                    cn.Open();
                    CustomerTable.Load(cmd.ExecuteReader());
                }
            }
        }
    }
}
